<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="odom_frame" default="odom" />
  <arg name="base_frame" default="base_footprint" />
  <arg name="global_frame" default="map" />

  <arg name="folder_name" default="move_base" />

  <arg name="cmd_topic" default="cmd_vel" />

  <!-- <arg name="footprint" default="[ [1.6, 0.5], [-0.4, 0.5], [-0.4, -0.5], [1.6, -0.5] ]" /> real size -->
  <!-- extended size -->
  <!-- <arg name="footprint" default="[ [1.7, 0.6], [-0.5, 0.6], [-0.5, -0.6], [1.7, -0.6] ]" />   -->

<!-- asd -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">

    <rosparam file="$(find tigra_software)/config/$(arg folder_name)/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find tigra_software)/config/$(arg folder_name)/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find tigra_software)/config/$(arg folder_name)/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find tigra_software)/config/$(arg folder_name)/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find tigra_software)/config/$(arg folder_name)/move_base_params.yaml" command="load" />

    <rosparam file="$(find tigra_software)/config/$(arg folder_name)/global_planner_params.yaml" command="load" />
    <rosparam file="$(find tigra_software)/config/$(arg folder_name)/teb_local_planner.yaml" command="load" />

    <!-- Explicit params setting -->
   <!-- <rosparam param="TebLocalPlannerROS/footprint_model/vertices" subst_value="True">$(arg footprint)</rosparam> -->
    <!-- <rosparam param="global_costmap/footprint" subst_value="True">$(arg footprint)</rosparam> -->
    <!-- <rosparam param="local_costmap/footprint" subst_value="True">$(arg footprint)</rosparam> -->

    <param name="global_costmap/global_frame" value="$(arg global_frame)" />
    <param name="global_costmap/robot_base_frame" value="$(arg base_frame)" />
    <param name="local_costmap/global_frame" value="$(arg odom_frame)" />
    <param name="local_costmap/robot_base_frame" value="$(arg base_frame)" />

    <param name="base_global_planner" value="global_planner/GlobalPlanner" />
    <param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS" />
    <param name="clearing_rotation_allowed" value="false" /> 

    <remap from="cmd_vel" to="$(arg cmd_topic)" />
  </node>


</launch> 
